src/components/MapView/MapView.tsx

```diff
@@ -0,0 +1,96 @@
+import {View} from 'react-native';
+import {useIsFocused} from '@react-navigation/native';
+import Mapbox, {MapState, MarkerView, setAccessToken} from '@rnmapbox/maps';
+import {forwardRef, useEffect, useImperativeHandle, useRef, useState} from 'react';
+
+import utils from './utils';
+import Direction from './Direction';
+import CONST from '../../CONST';
+
+import {MapViewProps, MapViewHandle} from './MapViewTypes';
+
+const MapView = forwardRef<MapViewHandle, MapViewProps>(({accessToken, style, mapPadding, styleURL, pitchEnabled, initialState, waypoints, directionCoordinates, directionStyle}, ref) => {
+    const isFocused = useIsFocused();
+    const cameraRef = useRef<Mapbox.Camera>(null);
+    const [isIdle, setIsIdle] = useState(false);
+
+    useImperativeHandle(
+        ref,
+        () => ({
+            flyTo: (location: [number, number], zoomLevel: number = CONST.MAPBOX.DEFAULT_ZOOM, animationDuration?: number) =>
+                cameraRef.current?.setCamera({zoomLevel, centerCoordinate: location, animationDuration}),
+            fitBounds: (ne: [number, number], sw: [number, number], paddingConfig?: number | number[] | undefined, animationDuration?: number | undefined) =>
+                cameraRef.current?.fitBounds(ne, sw, paddingConfig, animationDuration),
+        }),
+        [],
+    );
+
+    useEffect(() => {
```

Comment:
Would you mind adding some comments around these effects so that it's clear what their purpose is and why they are there (same for below)?

Reply:
Updated